src/OrchardCore.Modules/OrchardCore.OpenId/Drivers/OpenIdClientSettingsDisplayDriver.cs

```diff
@@ -205,11 +200,7 @@ public override async Task<IDisplayResult> UpdateAsync(OpenIdClientSettings sett
                     }
                 }
 
-                // If the settings are valid, release the current tenant.
-                if (context.Updater.ModelState.IsValid)
-                {
-                    await _shellHost.ReleaseShellContextAsync(_shellSettings);
-                }
+                site.QueueReleaseShellContext();
```

Comment:
As the bot also pointed out: why restart the tenant on any update? Also in `ReverseProxySettingsDisplayDriver` and others.

Reply:
Sometimes we update important settings that are loaded and saved in memory. Reloading the tenant will ensure that all the services are loading the correct values again.

Reply:
I understand that, but why update even if the model is invalid? This change changes that.

Reply:
In the case of settings, we do a single check in the controller and prevent the restart if the model state is invalid. There is no need to duplicate that logic in every driver where we already do the same thing in a centralized place “i.e., controller)

Reply:
This couples the drivers quite tightly to the controller. I don't think that's a safe approach. I get that this reduces the code and perhaps can also have some tiny performance benefit, but such approaches are really only good if you keep the whole thing in your head but can break if the next developer (and reviewer) has a narrower view of the codebase.

Reply:
Why is it up to the controller to decide if a restart is applicable?

Reply:
OK, good then.